utils/build-apple-framework.sh

```diff
@@ -56,11 +59,16 @@ function configure_apple_framework {
   else
     build_cli_tools="false"
   fi
+  if [[ $1 == catalyst ]]; then
```

Comment:
ditto

Reply:
Instead of using `platform`, I'd rephrase it to `HERMES_APPLE_TARGET_PLATFORM`, since that's what we are using this variable for. To change the value of `HERMES_APPLE_TARGET_PLATFORM` from `$1` when it's "catalyst", since that's not compatible with CMake.

However, using that as an argument and platform name is very clever - it gives us separate build folder and cache for both `macosx` and `catalyst`. If you look at the line `83` when we actually run `build` command, the build output is going to be `build_catalyst` or `build_macosx` since we're using `$1`.

If we replaced that to `$platform`, it would overwrite the build of another command.